.github/workflows/release-publish.yaml

```
@@ -0,0 +1,66 @@
+name: Release
+on:
+  push:
+    tags:
+      # These tags should be protected, remember to enable the rule:
+      # https://github.com/canonical/craft-application/settings/tag_protection
+      - "[0-9]+.[0-9]+.[0-9]+"
+
+permissions:
+  contents: write
+
+jobs:
+  source-wheel:
+    runs-on: [self-hosted, jammy]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Fetch tag annotations
+        run: |
+          git fetch --force --tags --depth 1
+          git describe --dirty --long --match '[0-9]*.[0-9]*.[0-9]*' --exclude '*[^0-9.]*'
+      - name: Setup Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.10"
+          check-latest: true
+      - name: Build packages
+        run: |
+          pip install build twine
```

Shouldn't be better to place it in `release` optional dependencies of the project?

---

but then wouldn't we need to do the building ourselves?

---

I just wanted to suggest moving `build` and `twine` to `pyproject.toml` specific optional-dependency section and replace above line with. Rest of code would stay the same - building would be still done using `python3 -m build`.
```suggestion
          pip install '.[release]'
```
It would allow to avoid introducing "hidden" dependencies in workflow directly.

---

I see what you mean but I don't want to change this too much (in comparison with starbase's)
maybe we should use [hynek/build-and-inspect-python-package](https://github.com/hynek/build-and-inspect-python-package?tab=readme-ov-file)?